katdal/test/test_chunkstore.py

```diff
@@ -91,6 +91,20 @@ def test_max_dim_elements_ignore(self):
         assert_equal(chunks, ((10,), 1024 * (8,), (144,)))
 
 
+def test_prune_chunks():
+    """Test the `_prune_chunks` internal function."""
+    chunks = ((10, 10, 10, 10), (2, 2, 2), (40,))
+    # The chunk start-stop boundaries on each axis are:
+    # ((0, 10, 20, 30, 40), (0, 2, 4, 6), (0, 40))
+    index = np.s_[13:34, :4, 10:]
+    new_chunks, new_index, new_offset = _prune_chunks(chunks, index)
+    # The new chunk start-stop boundaries on each axis are:
+    # ((10, 20, 30, 40), (0, 2, 4), (0, 40))
+    assert_equal(new_chunks, ((10, 10, 10), (2, 2), (40,)))
+    assert_equal(new_index, np.s_[3:24, 0:4, 10:40])
+    assert_equal(new_offset, (10, 0, 0))
+
```

Comment:
You could also add a test here for the case of a non-unit step for the slice.

Reply:
Good idea, done!